### ISSUE TITLE
Query with param as the last symbol crash the query

### ISSUE THREAD

**Issue by senior-sigan** [yandex/clickhouse-jdbc#259]
Version: 0.1.43.
JDK: 1.8.
---

When a query ends with a query parameter, the parameter is deleted from the query and it'll cause to an error on the clickhouse side.

Example: 
```
SELECT * FROM decisions  PREWHERE userID = ? AND eventDate >= toDate(?) AND eventDate <= toDate(?)  ORDER BY time DESC LIMIT ?, ?
```

In the `ClickHousePreparedStatementImpl#buildSql` the last `?` will be lost, thus this query will be generated: 
```
SELECT * FROM decisions  PREWHERE userID = '1c4b9dae-c1b0-4f1b-9710-a0ffa11b4898' AND eventDate >= toDate('2018-11-19') AND eventDate <= toDate('2018-11-20')  ORDER BY time DESC LIMIT 0,
```
Obviously, the last `,` symbol will crash the query and generate exception:
```
 Caused by: org.springframework.jdbc.UncategorizedSQLException: PreparedStatementCallback; uncategorized SQLException for SQL [SELECT * FROM decisions  PREWHERE userID = ? AND eventDate >= toDate(?) AND eventDate <= toDate(?)  ORDER BY time DESC LIMIT ?, ?]; SQL state [null]; error code [62]; ClickHouse exception, code: 62, host: audit.local, port: 8123; Code: 62, e.displayText() = DB::Exception: Syntax error: failed at position 188: FORMAT TabSeparatedWithNamesAndTypes;. Expected number, e.what() = DB::Exception
```

Workaround: add semicolon symbol: `;` to the end of the query. This will solve the problem. 

But I got this bug, when I'd updated from the `0.1.38` to the `0.1.43` version of the jdbc driver. So this behaviour should be either documented or fixed, I suppose.

Thanks.

**Comment by enqueue** [comment]
Sorry, I probably broke this :-(

**Comment by senior-sigan** [comment]
If the PR was accepted, I hope I'll see this fix in the next version.
For now I use the workaround with semicolon at the end. 
Thanks.

**Comment by serebrserg** [comment]
This must be available in 0.1.44, which was published today.
